hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/OmKeyLocationInfoGroup.java

```diff
@@ -143,16 +143,22 @@ public static OmKeyLocationInfoGroup getFromProtobuf(
   }
 
   /**
-   * Given a new block location, generate a new version list based upon this
-   * one.
+   * Given a new block location, generate a new version list based upon the
+   * version number of this object. Keep the blocks to when keepVersions is
+   * on for object versioning.
    *
    * @param newLocationList a list of new location to be added.
+   * @param keepVersions a knob to carry current locations to the next version.
    * @return newly generated OmKeyLocationInfoGroup
    */
   OmKeyLocationInfoGroup generateNextVersion(
-      List<OmKeyLocationInfo> newLocationList) {
-    Map<Long, List<OmKeyLocationInfo>> newMap =
-        new HashMap<>();
+      List<OmKeyLocationInfo> newLocationList, boolean keepVersions) {
+    Map<Long, List<OmKeyLocationInfo>> newMap = null;
+    if (keepVersions) {
+      newMap = new HashMap<>(locationVersionMap);
```

Comment:
We don‘t need to include the locationVersionMap to the new  OmKeyLocationInfoGroup even when version is true.  
The current OmKeyLocationInfoGroup design actually has one point to improve.   You can refer to @symious document https://docs.google.com/document/d/1UYVETpSHyVppNPjHq8VuXj2t2Y0LYQNRYa6awSTCROA/edit#heading=h.h2k2ovrsrw33 for more detail information.

Reply:
Thank you for the review. I updated the commented  part so that it does keep past location version map and also added option 3 of my idea in this issue to the @symious document.

I'm sorry that I don't have much time to add necessary unit tests and make integration tests passing for now, but will try in a few days.